api/src/task/sam_extracts/cleanup_old_sam_extracts.py

```diff
@@ -0,0 +1,112 @@
+"""Task to cleanup old SAM.gov extract files."""
+
+import logging
+from datetime import date, timedelta
+from enum import StrEnum
+from typing import Sequence
+
+from sqlalchemy import select
+
+import src.adapters.db as db
+from src.constants.lookup_constants import SamGovProcessingStatus
+from src.db.models.sam_extract_models import SamExtractFile
+from src.task.task import Task
+from src.util import file_util
+
+logger = logging.getLogger(__name__)
+
+
+class CleanupOldSamExtractsTask(Task):
+    """Task that runs daily to cleanup old SAM.gov extract files"""
+
+    class Metrics(StrEnum):
+        FILES_DELETED_COUNT = "files_deleted_count"
+        ERRORS_ENCOUNTERED = "errors_encountered"
+
+    def __init__(self, db_session: db.Session) -> None:
+        super().__init__(db_session)
+
+    def run_task(self) -> None:
+        """Main task logic to cleanup old SAM.gov extract files"""
+        logger.info("Starting cleanup of old SAM.gov extract files")
+
+        # Get files older than 45 days that aren't already deleted
+        old_files = self._get_old_files_to_cleanup()
+
+        if not old_files:
+            logger.info("No old files found to cleanup")
+            return
+
+        logger.info(f"Found {len(old_files)} old files to cleanup")
+
+        deleted_count = 0
+        error_count = 0
+
+        for sam_extract_file in old_files:
+            try:
+                self._cleanup_file(sam_extract_file)
+                deleted_count += 1
+                self.increment(self.Metrics.FILES_DELETED_COUNT)
+
+                logger.info(
+                    "Successfully cleaned up old SAM.gov extract file",
+                    extra={
+                        "sam_extract_file_id": sam_extract_file.sam_extract_file_id,
+                        "s3_path": sam_extract_file.s3_path,
+                        "extract_date": sam_extract_file.extract_date,
+                        "extract_type": sam_extract_file.extract_type,
+                    },
+                )
+
+            except Exception as e:
+                error_count += 1
+                self.increment(self.Metrics.ERRORS_ENCOUNTERED)
+
+                logger.error(
+                    "Failed to cleanup old SAM.gov extract file",
+                    extra={
+                        "sam_extract_file_id": sam_extract_file.sam_extract_file_id,
+                        "s3_path": sam_extract_file.s3_path,
+                        "extract_date": sam_extract_file.extract_date,
+                        "extract_type": sam_extract_file.extract_type,
+                        "error": str(e),
+                    },
+                    exc_info=True,
+                )
```

Comment:
I'd say if we hit an error, we just let the process fail - with what we're doing here being just a simple file delete, I think the only possible errors are permissions and AWS outages - so if we hit one, we probably just aren't getting anything done.

```suggestion
            except Exception:
                logger.exception(
                    "Failed to cleanup old SAM.gov extract file",
                    extra={
                        "sam_extract_file_id": sam_extract_file.sam_extract_file_id,
                        "s3_path": sam_extract_file.s3_path,
                        "extract_date": sam_extract_file.extract_date,
                        "extract_type": sam_extract_file.extract_type,
                    },
                )
                raise
```